koocook_core/models/review.py

```diff
@@ -123,30 +131,57 @@ class Rating(models.Model):
         null=True,
         blank=True,
     )
+    used = models.BooleanField(default=False, blank=True)
+
+    def __init__(self, *args, **kwargs):
+        kwargs = parse_kwargs_item_reviewed(kwargs, strict=False)
+        if 'rating_value' in kwargs:
+            if not kwargs.get('worst_rating', 1) <= \
+                   kwargs['rating_value'] <= \
+                   kwargs.get('best_rating', 5):
+                raise ValueError('`rating_value` must be between `best_rating` and '
+                                 '`worst_rating`')
+        super().__init__(*args, **kwargs)
 
     @property
     def item_reviewed(self):
         return self.reviewed_recipe or self.reviewed_post or self.reviewed_comment
 
-    @item_reviewed.setter
-    def item_reviewed(self, obj):
+    def save(self, *args, **kwargs):
+        if self.item_reviewed is None:
+            raise ValidationError(_('There must be at least 1 item reviewed'))
+        super().save(*args, **kwargs)
+
+
+def parse_kwargs_item_reviewed(kwargs: Dict[str, Any], strict: bool = True) -> Dict[str, Any]:
+    item = kwargs.pop('item_reviewed', None)
+    if item:
+        assert not any(
+            k in kwargs
+            for k in ('reviewed_recipe', 'reviewed_post', 'reviewed_comment')
+        ), "Don't specify both item reviewed and the actual item reviewed"
         from koocook_core.models.recipe import Recipe
         from koocook_core.models.post import Post
-        try:
-            if self.reviewed_recipe:
-                assert isinstance(obj, Recipe)
-                self.reviewed_recipe = obj
-            elif self.reviewed_post:
-                assert isinstance(obj, Post)
-                self.reviewed_post = obj
-            elif self.reviewed_comment:
-                assert isinstance(obj, Comment)
-                self.reviewed_comment = obj
-        except AssertionError as e:
-            raise TypeError('item_reviewed must be of the correct type '
-                            '\'{}\' not \'\''.format(
-                             type(self.reviewed_recipe or self.reviewed_post or self.reviewed_comment),
-                             type(obj))) from e.__context__
+        if isinstance(item, Recipe):
+            kwargs['reviewed_recipe'] = item
+        elif isinstance(item, Post):
+            kwargs['reviewed_post'] = item
+        elif isinstance(item, Comment):
+            kwargs['reviewed_comment'] = item
+        else:
+            raise TypeError(f'item_reviewed must be of the type Recipe, '
+                            f"Post or Comment not '{type(item)}'")
+    else:
+        # monkey patch
```

Comment:
Aiyah

Reply:
Regardless of the first 2 lines

Reply:
Somehow there's a use case calling this function from a models' `__init__()` with empty kwargs. I didn't know where it came from so I just put it there for that case specifically.

Reply:
Resolved in 7f95393764c2a9aab0e23b7a1eb1e0f5d0a38246